src/Storage/Storage/File/Cmdlet/StartAzureStorageFileCopy.cs

```diff
@@ -251,8 +251,9 @@ public override void ExecuteCmdlet()
         {
             if(this.DestFile != null)
             {
-                // when only track1 object input, will miss storage context, so need to build storage context for prepare the output object.
-                if (this.DestContext == null)
+                // Build and set storage context for the output object when
+                // 1. input track1 object and storage context is missing 2. the current context doesn't match the context of the input object 
+                if (ShouldSetContext(this.Context, this.DestFile.ServiceClient))
```

Comment:
should be "this.DestContext", instead of "this.Context"

Reply:
Done